### PR TITLE
Adjust upgrade strategy to latest upstream changes

### PR DESCRIPTION
**Pull request by jschmid1**
These commits respect some recent changes also:

  * ceph-mgr is now mandatory (adapt keyrings)
  * only restart mons first
  * set luminous osds
  * add a wait to allow ceph to catch up after a reboot
  * add preflight checks
  * make zypper-migration default
  * fix a sync bug

In addition we have keyring changes that needs to be reviewed please.
keyring files I haven't touched yet are: ganesha, benchmark. Could you please confirm that mgr capabilities have to be adjusted for them aswell?

### CODE REVIEW

File: srv/modules/runners/orderednodes.py [hunk 1]
```diff
@@ -15,6 +15,16 @@ def _preserve_order_sorted(seq):
     seen = set()
     return [x for x in seq if x not in seen and not seen.add(x)]
 
+def only(cluster='ceph', role=None):
```
Comment (jan--f) [inline srv/modules/runners/orderednodes.py]: Is the purpose of this function to get a list of all minions in the cluster?
If so we already have the select runner. A list of minions can be aquired using the ```select.minions``` runner call.

File: srv/salt/ceph/mds/files/keyring.j2 [hunk 2]
```diff
@@ -1,6 +1,7 @@
 [{{ client }}]
         key = {{ secret }}
-        caps mds = "allow"
-        caps mon = "allow profile mds"
-        caps osd = "allow rwx"
+        caps mds = "allow *"
```
Comment (jan--f) [inline srv/salt/ceph/mds/files/keyring.j2]: Please don't re-add the * for the mds caps.

File: srv/salt/ceph/wait/until/expired/30sec/init.sls [hunk 3]
```diff
@@ -0,0 +1,3 @@
+include:
+  - .{{ salt['pillar.get']('wait_until_ok', 'default') }}
```
Comment (jan--f) [inline srv/salt/ceph/wait/until/expired/30sec/init.sls]: This clashes with ```srv/salt/ceph/wait/until/OK/init.sls```. Is that intentional ?

File: srv/modules/runners/orderednodes.py (hunk 1 above)
Comment (jan--f) [inline srv/modules/runners/orderednodes.py]: Sorry of course the function can return a list of a certain role. Though I believe the select runner can do this as well via ```select.attr```?

File: srv/salt/ceph/mds/files/keyring.j2 (hunk 2 above)
Comment (jschmid1) [inline srv/salt/ceph/mds/files/keyring.j2]: Mh, we had the "allow *" caps in [SES4](https://github.com/SUSE/DeepSea/blob/SES4/srv/salt/ceph/mds/files/keyring.j2), which will not be refreshed if you upgrade to SES5.

```
ceph auth get mds.<mdsnode>:
        caps mds = "allow *"
        caps mgr = "allow profile mds"
        caps mon = "allow profile mds"
        caps osd = "allow rwx"
```
If you do a `ceph auth add mds.data1 -i /srv/salt/ceph/mds/cache/data1.keyring` if will fail with:

` Error EINVAL: entity mds.data1 exists but cap mds does not match`

Do we have to remove/reauth in order to get it right?

File: srv/salt/ceph/wait/until/expired/30sec/init.sls (hunk 3 above)
Comment (jschmid1) [inline srv/salt/ceph/wait/until/expired/30sec/init.sls]: good catch, c&p mistakes

File: srv/modules/runners/orderednodes.py (hunk 1 above)
Comment (jschmid1) [inline srv/modules/runners/orderednodes.py]: Right, I've missed that.

`salt-run select.minions roles=mon` would've worked too.

Updating

File: srv/salt/ceph/mds/files/keyring.j2 (hunk 2 above)
Comment (jan--f) [inline srv/salt/ceph/mds/files/keyring.j2]: Hmm true. And that issue doesn't exist when adding the mgr caps?

Comment (jschmid1) [inline srv/salt/ceph/mds/files/keyring.j2]: Well, it doesn't exist if the provided keyring *exactly* matches the previous keyring(including the *)

**Comment by jschmid1** [comment]
closes #450

**Pull request merged**